### ISSUE TITLE
Make `dat sync` consistent with other commands

### ISSUE THREAD

**Issue by mafintosh** [datproject/dat#784]
The `dat <link> <dest>` is extremely useful when dealing with dats on servers since it is idempotent, but noticed we haven't documented it anywhere

Either we should document it or make `sync` do the same thing.

Currently the documented approaches forces you to clone first which disables upload and implies that you know what state the dat was in before you ran the command, which is tricky when you just wanna automate syncing

**Comment by mafintosh** [comment]
I'd suggest, we change it to do the following:

1. Have `dat sync` behave exactly as `dat <link>`
1. Have `dat clone` be the same as `dat sync` but exit once done
1. Have `dat pull` be the same as `dat sync` but exit once done and fail if the dat is not there

Introduce `--no-upload` and `--no-download` to disable upload and download.

**Comment by joehand** [comment]
Definitely need to document `dat <link> <dest>` and `dat <dir>`! I'll add that to the usage now. Still in the process of updating readme docs but I'll put that on the list.

> I'd suggest, we change it to do the following:
> 1. Have `dat sync` behave exactly as `dat <link>`
> 2. Have `dat clone` be the same as `dat sync` but exit once done, and fail if it cannot find peers
> 3. Have `dat pull` be the same as `dat sync` but exit once done, fail if it cannot find peers, and fail if the dat is not there

Sounds good! This is how it's supposed to work, are these commands not doing that?

> Introduce --no-upload and --no-download to disable upload and download.

I turned on the `--upload` option in `clone` and `pull` to fix #783. But I can make these global options. We should separate out the announce from upload as well (https://github.com/datproject/dat-node/issues/150)

> Both clone and pull should call .leave on discovery-swarm when done as well to announce from the disc network

I can add this, but not quite sure what this means =).

**Comment by joehand** [comment]
> i cannot run "dat sync <link> <dest>" if i havent cloned the dat

**Comment by joehand** [comment]
tldr; dat sync needs to run share/create directly. 

We can probably get rid of the `src/commands/sync.js` totally. It'd help make options more consistent.

**Comment by mitar** [comment]
> Currently the documented approaches forces you to clone first which disables upload

```
dat clone --help
Clone a remote Dat archive

Usage: dat clone <link> [download-folder]
    --dir, -d             set the directory for Dat
    --port                port to use for connections (default: 3282)
    --utp                 use utp for discovery (default: true)
    --http                serve dat over http (default port: 8080)
    --sparse              download only requested data (default: false)
    --up                  throttle upload bandwidth (1024, 1kb, 2mb, etc.)
    --down                throttle download bandwidth (1024, 1kb, 2mb, etc.)
    --empty               Do not download files by default. Files must be synced manually. (default: false)
    --upload              announce your address on link (improves connection capability) and upload data to other downloaders. (default: true)
    --show-key, -k        print out the dat key (default: false)
```

It seems it says that `--upload` is on by default?

**Comment by okdistribute** [comment]
This will be fixed by #1173
